provider/testdata/src/lib.rs

```diff
@@ -17,51 +17,41 @@
 //!
 //! ```
 //! use icu::locid::locale;
-//! use icu_provider::hello_world::*;
-//! use icu_provider::prelude::*;
+//! use icu::list::{ListFormatter, ListLength};
 //!
-//! let req = DataRequest {
-//!     locale: &locale!("en").into(),
-//!     metadata: Default::default(),
-//! };
+//! // Unstable constructor
+//! ListFormatter::try_new_and_with_length_unstable(
```

Comment:
Issue: I'd _prefer_ to not promote any particular component to be the one we use for docs tests of other things. HelloWorld is appropriate; it's dead simple, and it's clear that the thing you care about is not the HelloWorld.

Reply:
Ok, but then this should use `HelloWorldFormatter`. Clients should not need to know about how to use the data provider traits to manually load data.

Reply:
I'm fine changing the example to use `HelloWorldFormatter` with the three constructor types (unstable, any, buffer).